app/stac_api/models.py

```diff
@@ -656,6 +531,12 @@ def from_db(cls, db, field_names, values):
     def __str__(self):
         return self.name
 
+    def clean(self):
+        # as the media type sometimes is with space, sometimes without
+        # it will alway be stored without
+        if self.media_type:
+            self.media_type = self.media_type.replace(' ', '')
```

Comment:
This is a field only clean step and should be placed in `clean_media_type()` method (see https://docs.djangoproject.com/en/3.1/ref/forms/validation/#form-and-field-validation)